EmailValidator/Validation/MailboxCheckValidation.php

```diff
@@ -0,0 +1,240 @@
+<?php
+
+namespace Egulias\EmailValidator\Validation;
+
+use Egulias\EmailValidator\EmailLexer;
+use Egulias\EmailValidator\Exception\InvalidEmail;
+use Egulias\EmailValidator\Validation\Error\IllegalMailbox;
+use Egulias\EmailValidator\Warning\NoDNSMXRecord;
+use Egulias\EmailValidator\Warning\SocketWarning;
+use Egulias\EmailValidator\Warning\Warning;
+
+class MailboxCheckValidation implements EmailValidation
+{
+    const END_OF_LINE = "\r\n";
+
+    /**
+     * @var InvalidEmail
+     */
+    private $error;
+
+    /**
+     * @var Warning[]
+     */
+    private $warnings = [];
+
+    /**
+     * @var int
+     */
+    private $lastResponseCode;
+
+    /**
+     * @var int
+     */
+    private $port = 25;
+
+    /**
+     * @var int
+     */
+    private $timeout = 10;
+
+    /**
+     * @var string
+     */
+    private $fromEmail = 'test-mailbox@validation.email';
+
+    /**
+     * MailboxCheckValidation constructor.
+     */
+    public function __construct()
+    {
+        if (!extension_loaded('intl')) {
+            throw new \LogicException(sprintf('The %s class requires the Intl extension.', __CLASS__));
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getError()
+    {
+        return $this->error;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getWarnings()
+    {
+        return $this->warnings;
+    }
+
+    /**
+     * @return int
+     */
+    public function getLastResponseCode()
+    {
+        return $this->lastResponseCode;
+    }
+
+    /**
+     * @return int
+     */
+    public function getPort()
+    {
+        return $this->port;
+    }
+
+    /**
+     * @param int $port
+     */
+    public function setPort($port)
+    {
+        $this->port = $port;
+    }
+
+    /**
+     * @return int
+     */
+    public function getTimeout()
+    {
+        return $this->timeout;
+    }
+
+    /**
+     * @param int $timeout
+     */
+    public function setTimeout($timeout)
+    {
+        $this->timeout = $timeout;
+    }
+
+    /**
+     * @return string
+     */
+    public function getFromEmail()
+    {
+        return $this->fromEmail;
+    }
+
+    /**
+     * @param string $fromEmail
+     */
+    public function setFromEmail($fromEmail)
+    {
+        $this->fromEmail = $fromEmail;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function isValid($email, EmailLexer $emailLexer)
+    {
+        $mxHosts = $this->getMXHosts($email);
+
+        $isValid = false;
+        foreach ($mxHosts as $mxHost) {
+            if ( ($isValid = $this->checkMailbox($mxHost, $this->port, $this->timeout, $this->fromEmail, $email)) ) {
+                break;
```

Comment:
For the sake of clarity, please change to
```php
if($this->checkMailbox(...)) {
  $isValid = true;
  break;
}
```

Reply:
fix